flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/source/ContinuousFileReaderOperator.java

```diff
@@ -334,9 +342,11 @@ public void run() {
 				this.pendingSplits.remove();
 			}
 
-			if (this.format instanceof CheckpointableInputFormat && this.isSplitOpen) {
-				S formatState = (S) ((CheckpointableInputFormat) format).getCurrentState();
-				return new Tuple3<>(snapshot, currentSplit, currentSplit == null ? null : formatState);
+			if (this.format instanceof CheckpointableInputFormat && this.currentSplit != null) {
+				S formatState = this.isSplitOpen ?
+					(S) ((CheckpointableInputFormat) format).getCurrentState() :
+					restoredFormatState;
+				return new Tuple3<>(snapshot, currentSplit, formatState);
 			} else {
 				LOG.info("The format used is not checkpointable. The current input split will be restarted upon recovery.");
```

Comment:
I think this log message is wrong for `this.currentSplit != null`. It leads to pollution of log files.